### ISSUE TITLE
Merge "Doctrine Event Listeners and Subscribers" and "Entity Listeners"?

### ISSUE THREAD

**Issue by ThomasLandauer** [symfony/symfony-docs#10198]
Following up on https://github.com/symfony/symfony-docs/pull/10195

I'm suggesting to merge https://symfony.com/doc/current/bundles/DoctrineBundle/entity-listeners.html and http://symfony.com/doc/current/doctrine/event_listeners_subscribers.html#creating-the-listener-class

Reason: Shouldn't the entity listeners be presented as the preferred way of doing it?
The note that they where introduced in Doctrine 2.4 suggests that they're a brand new feature - however, Doctrine 2.4 was released some 5 years ago ;-)

So since entity listeners have been around for some time, what's the use case for event listeners and subscribers at all anymore?

**Comment by weaverryan** [comment]
I agree! These should certainly live in just one doc. However, two important things:

1) The DoctrineBundle docs (even though they are displayed on symfony.com) are not in our control - they develop those docs separately from us in their repository. For sure we should try to make them work together as much as possible. What we should do first is make our docs how we want them. Then, if there is some duplication, we can propose to remove some things from the bundle.

2) Unrelated to the docs, I believe that the`doctrine.orm.entity_listener` tag is NOT currently autoconfigured. We should really change that in the bundle if possible :)

**Comment by TomasVotruba** [comment]
> @weaverryan Entity listeners do not implement a common interface. That's why they cannot be autoconfigured.

Sure, but why not subscribers that implement `EventSubscriber` interface? 

```yaml
services:
    # why this is not autoconfigured? :(
    Pehapkari\Doctrine\EventSubscriber\SetUploadDestinationOnPostLoadEventSubscriber:
        tags:
            - { name: doctrine.event_subscriber }
```

Now I have to do this manually. It's really WTF while all other common interfaces autoconfigured.

**Comment by xabbuh** [comment]
IIRC the issue with auto configuring `EventSubscriberInterface` instances is that you do not know if the implementing class is an event subscriber for the ORM or any of the other implementations.

**Comment by weaverryan** [comment]
That is a problem - but one that could be overcome by adding a sub interface to the bundle - we’ve done that for fixtures. The problem is multiple entity managers. We would need to autoconfigure to the default one. If a user didn’t want that, they would need to add a tag manually with config for the other entity manager. That’s not a problem, but figuring out what’s going on in the compiler pass is challenging. Should the second tag override the first? Or does the user actually want listeners on both entity managers? 

It’s a weird edge case that shouldn’t stop us from implementing autoconfigure, but it has so far. One idea is, for autoconfigure, add the tag *and* a special attribute - something like from_autoconfigure. Then, if the compiler pass detects 2 tags, but one has this attribute, it skips that one. It’s a clear way for us to know that the autoconfigure tag has been “overridden”.

**Comment by OskarStark** [comment]
Sounds like a straightforward idea from DX POV Ryan 👍🏻

**Comment by javiereguiluz** [comment]
Closing as "fixed" because we've merged all articles about Doctrine events into a single article: https://symfony.com/doc/current/doctrine/events.html  The doc in the DoctrineBundle will be kept because all bundles provide some basic doc. 

If we disagree, we can open an issue in the DoctrineBundle repo asking for the removal of their doc and to link instead to the new article in Symfony Docs. 

Thanks.
